### PR TITLE
Implements callback & sign require Base64 document

### PR DESCRIPTION
**Pull request by Gwemox**

### CODE REVIEW

File: Requester/XmlRpcRequester.php [hunk 1]
```diff
@@ -71,7 +72,7 @@ public static function flatten($data, bool $skipNullValue = true)
 
         if (is_object($data) &&
             !($data instanceof \Laminas\XmlRpc\Value\DateTime) &&
-            !($data instanceof \Laminas\XmlRpc\Value\Base64)
+            !($data instanceof Base64)
```
Comment (konandrum) [inline Requester/XmlRpcRequester.php:76]: Keep full FQCN

File: Requester/XmlRpcRequester.php [hunk 2]
```diff
@@ -193,14 +194,12 @@ public function requestTransaction(TransactionRequest $transactionRequest): Tran
             $transactionResponse
                 ->setId($response['id'])
                 ->setUrl($response['url'])
-                ->setState(TransactionResponse::STATE_SUCCESS)
-            ;
+                ->setState(TransactionResponse::STATE_SUCCESS);
```
Comment (konandrum) [inline Requester/XmlRpcRequester.php]: syntax

File: Requester/XmlRpcRequester.php [hunk 3]
```diff
@@ -193,14 +194,12 @@ public function requestTransaction(TransactionRequest $transactionRequest): Tran
...
         } catch (FaultException $fe) {
             $transactionResponse
                 ->setState(TransactionResponse::STATE_ERROR)
                 ->setErrorCode($fe->getCode())
-                ->setErrorMessage($fe->getMessage())
-            ;
+                ->setErrorMessage($fe->getMessage());
```
Comment (konandrum) [inline Requester/XmlRpcRequester.php]: syntax

File: Requester/XmlRpcRequester.php [hunk 4]
```diff
@@ -263,8 +262,7 @@ public function getTransactionInfo(string $transactionId): TransactionInfo
                 ->setCustomerId($response['customerId'] ?? null)
                 ->setTransactionId($response['transactionId'] ?? null)
                 ->setRedirectPolicy($response['redirectPolicy'] ?? null)
-                ->setRedirectWait($response['redirectWait'] ?? null)
-            ;
+                ->setRedirectWait($response['redirectWait'] ?? null);
```
Comment (konandrum) [inline Requester/XmlRpcRequester.php]: syntax

File: Requester/XmlRpcRequester.php [hunk 5]
```diff
@@ -274,8 +272,7 @@ public function getTransactionInfo(string $transactionId): TransactionInfo
             $transactionInfo
                 ->setState(TransactionInfo::STATE_ERROR)
                 ->setErrorCode($fe->getCode())
-                ->setErrorMessage($fe->getMessage())
-            ;
+                ->setErrorMessage($fe->getMessage());
```
Comment (konandrum) [inline Requester/XmlRpcRequester.php]: syntax

File: Requester/XmlRpcRequester.php [hunk 6]
```diff
@@ -302,16 +299,14 @@ public function getTransactionInfoByCustomId(string $customId): TransactionInfo
                 ->setCustomerId($response['customerId'] ?? null)
                 ->setTransactionId($response['transactionId'] ?? null)
                 ->setRedirectPolicy($response['redirectPolicy'] ?? null)
-                ->setRedirectWait($response['redirectWait'] ?? null)
-            ;
+                ->setRedirectWait($response['redirectWait'] ?? null);
```
Comment (konandrum) [inline Requester/XmlRpcRequester.php]: syntax

File: Requester/XmlRpcRequester.php [hunk 7]
```diff
@@ -302,16 +299,14 @@ public function getTransactionInfoByCustomId(string $customId): TransactionInfo
...
         } catch (FaultException $fe) {
             $this->logger->error(sprintf('[Universign - requester.getTransactionInfoByCustomId] ERROR (%s): %s', $fe->getCode(), $fe->getMessage()));
 
             $transactionInfo
                 ->setState(TransactionInfo::STATE_ERROR)
                 ->setErrorCode($fe->getCode())
-                ->setErrorMessage($fe->getMessage())
-            ;
+                ->setErrorMessage($fe->getMessage());
```
Comment (konandrum) [inline Requester/XmlRpcRequester.php]: syntax

File: Requester/XmlRpcRequester.php [hunk 8]
```diff
@@ -328,16 +323,14 @@ public function relaunchTransaction(string $transactionId): TransactionInfo
             $response = $this->xmlRpcClient->call('requester.relaunchTransaction', $transactionId);
             $this->logger->info('[Universign - requester.relaunchTransaction] SUCCESS');
             $transactionInfo
-                ->setState(TransactionInfo::STATE_SUCCESS)
-            ;
+                ->setState(TransactionInfo::STATE_SUCCESS);
```
Comment (konandrum) [inline Requester/XmlRpcRequester.php]: syntax

File: Requester/XmlRpcRequester.php [hunk 9]
```diff
@@ -328,16 +323,14 @@ public function relaunchTransaction(string $transactionId): TransactionInfo
...
         } catch (FaultException $fe) {
             $this->logger->error(sprintf('[Universign - requester.relaunchTransaction] ERROR (%s): %s', $fe->getCode(), $fe->getMessage()));
 
             $transactionInfo
                 ->setState(TransactionInfo::STATE_ERROR)
                 ->setErrorCode($fe->getCode())
-                ->setErrorMessage($fe->getMessage())
-            ;
+                ->setErrorMessage($fe->getMessage());
```
Comment (konandrum) [inline Requester/XmlRpcRequester.php]: syntax

File: Requester/XmlRpcRequester.php [hunk 10]
```diff
@@ -354,16 +347,14 @@ public function cancelTransaction(string $transactionId): TransactionInfo
             $response = $this->xmlRpcClient->call('requester.cancelTransaction', $transactionId);
             $this->logger->info('[Universign - requester.cancelTransaction] SUCCESS');
             $transactionInfo
-                ->setState(TransactionInfo::STATE_SUCCESS)
-            ;
+                ->setState(TransactionInfo::STATE_SUCCESS);
```
Comment (konandrum) [inline Requester/XmlRpcRequester.php]: syntax

File: Requester/XmlRpcRequester.php [hunk 11]
```diff
@@ -354,16 +347,14 @@ public function cancelTransaction(string $transactionId): TransactionInfo
...
         } catch (FaultException $fe) {
             $this->logger->error(sprintf('[Universign - requester.cancelTransaction] ERROR (%s): %s', $fe->getCode(), $fe->getMessage()));
 
             $transactionInfo
                 ->setState(TransactionInfo::STATE_ERROR)
                 ->setErrorCode($fe->getCode())
-                ->setErrorMessage($fe->getMessage())
-            ;
+                ->setErrorMessage($fe->getMessage());
```
Comment (konandrum) [inline Requester/XmlRpcRequester.php]: syntax

**Review by konandrum — CHANGES_REQUESTED** [review]

File: Requester/XmlRpcRequester.php (hunk 9 above)
Comment (Gwemox) [reply Requester/XmlRpcRequester.php]: Fix

File: Requester/XmlRpcRequester.php (hunk 10 above)
Comment (Gwemox) [reply Requester/XmlRpcRequester.php]: Fix

File: Requester/XmlRpcRequester.php (hunk 11 above)
Comment (Gwemox) [reply Requester/XmlRpcRequester.php]: Fix

File: Requester/XmlRpcRequester.php (hunk 8 above)
Comment (Gwemox) [reply Requester/XmlRpcRequester.php]: Fix

File: Requester/XmlRpcRequester.php (hunk 7 above)
Comment (Gwemox) [reply Requester/XmlRpcRequester.php]: Fix

File: Requester/XmlRpcRequester.php (hunk 5 above)
Comment (Gwemox) [reply Requester/XmlRpcRequester.php]: Fix

File: Requester/XmlRpcRequester.php (hunk 4 above)
Comment (Gwemox) [reply Requester/XmlRpcRequester.php]: Fix

File: Requester/XmlRpcRequester.php (hunk 3 above)
Comment (Gwemox) [reply Requester/XmlRpcRequester.php]: Fix

File: Requester/XmlRpcRequester.php (hunk 2 above)
Comment (Gwemox) [reply Requester/XmlRpcRequester.php]: Fix

File: Requester/XmlRpcRequester.php (hunk 1 above)
Comment (Gwemox) [reply Requester/XmlRpcRequester.php:76]: Fix

File: Event/UniversignCallbackEvent.php [hunk 12]
```diff
@@ -4,22 +4,41 @@
...
-    const INVALID = 'INVALID';
-    const TRANSACTION_VALID = 'mpp_universign.callback.valid';
-    const TRANSACTION_INVALID = 'mpp_universign.callback.invalid';
+    const STATUS_READY = 0;
+    const STATUS_EXPIRED = 1;
+    const STATUS_COMPLETED = 2;
+    const STATUS_CANCELED = 3;
+    const STATUS_ERROR = 4;
+    /**
+     * All signatories have signed BUT awaiting validation by the Universign registration authority. All signatories have signed.
```
Comment (konandrum) [inline Event/UniversignCallbackEvent.php]:
```suggestion
     * All signatories have signed BUT waiting for Universign registration authority validation.
```

**Review by konandrum — CHANGES_REQUESTED** [review]

**Pull request merged**